bpf: split complexity configurations into separate files

lmb opened the pull request:
The complexity testsuite currently loads configuration from a single file for each BPF object and kernel version. In the file each line is a distinct configuration for that BPF object. For example:

    -DFOO=1
    -DBAR=1

would compile twice. Once with FOO defined, once with BAR. In practice we have about 30 to 40 defines per each line, which makes editing, diffing, etc. a nightmare.

Switch the format from one line per configration to one file per configuration. This makes it easier to diff and also to resolve merge conflicts.

The existing configurations were converted to invidiual files using the following script:

    #!/bin/bash

    set -euo pipefail

    for dir in bpf/complexity-tests/*; do
        test -d "$dir" || continue
        pushd "$dir" || exit 2
        for f in *.txt; do
            i=0
            while read -r line; do
                test -z "$line" && continue
                i=$((i + 1))
                target="$(basename "$f" .txt)"
                mkdir -p "$target"
                echo "${line// /$'\n'}" > "$target/$i.txt"
            done < "$f"
        done
        popd || exit 2
    done

dylandreimerink commented on test/verifier/verifier_test.go:
```diff
@@ -62,10 +63,36 @@ func getCIKernelVersion(t *testing.T) (string, string) {
+func readDatapathConfig(t *testing.T, r io.Reader) string {
+	scanner := bufio.NewScanner(r)
+	var lines []string
+	for i := 1; scanner.Scan(); i++ {
```
nit: `i` doesn't seem to be used, this can just be `for scanner.Scan() {`

dylandreimerink reviewed (approved):
LGTM!

lmb committed:
bpf: split complexity configurations into separate files

The complexity testsuite currently loads configuration from
a single file for each BPF object and kernel version. In the
file each line is a distinct configuration for that BPF object.
For example:

    -DFOO=1
    -DBAR=1

would compile twice. Once with FOO defined, once with BAR.
In practice we have about 30 to 40 defines per each line,
which makes editing, diffing, etc. a nightmare.

Switch the format from one line per configuration to one
file per configuration. This makes it easier to diff and
also to resolve merge conflicts.

The existing configurations were converted to invidiual
files using the following script:

    #!/bin/bash

    set -euo pipefail

    for dir in bpf/complexity-tests/*; do
        test -d "$dir" || continue
        pushd "$dir" || exit 2
        for f in *.txt; do
            i=0
            while read -r line; do
                test -z "$line" && continue
                i=$((i + 1))
                target="$(basename "$f" .txt)"
                mkdir -p "$target"
                echo "${line// /$'\n'}" > "$target/$i.txt"
            done < "$f"
        done
        popd || exit 2
    done

Signed-off-by: Lorenz Bauer <lmb@isovalent.com>
